models/job.go

```diff
@@ -279,6 +279,8 @@ type JobService interface {
 	GetByName(string, NamespaceSpec) (JobSpec, error)
 	// Dump returns the compiled Job
 	Dump(NamespaceSpec, JobSpec) (Job, error)
+	// Replay replays the jobSpec and its dependencies between start and endDate
+	Replay(NamespaceSpec, JobSpec, bool, time.Time, time.Time) (*DAGNode, error)
```

Comment:
I am not sure if the response of this interface method that is `DAGNode` fits here. I mean, replay is a lot more than just the execution tree.

Reply:
updated

Reply:
for now since we just have dry run functionality, lets just send the execution tree. Later we will add other values

Reply:
What do you have in mind? So are we going to have two functions like `ReplayDryRun` and `Replay` or just `Replay` with a composite response that will have a field `DryRun` true along with a execution tree? The thing is this is also indirectly affecting your `proto` for replay, and even though its easy to do changes in `interface`, we will have to do breaking changes in proto as well.

Reply:
made this change